server/lib/sync-agent.js

```diff
@@ -192,7 +198,15 @@ export default class SyncAgent {
     }
     return this.streamQuery(query, options)
       .then(stream => this.sync(stream, started_sync_at))
+      .then(result => {
+        // we weren't closing client after the stream was complete
+        // should close and continue to return result
+        this.closeClient();
```

Comment:
We weren't calling closeClient before at the end of any stream... seems like that could be leaking a lot of stuff...